Various fixes/missing fields

florian1345 opened the pull request:
Just an informal collection of problems I encountered and fixed locally.
Sorry for not presenting it in a more organized way.
Feel free to modify/integrate it however you like.

- Added `name` field to `TypeDefinition` nodes
- Added `isVirtual` and `isOverride` fields to `ModifierDefinition` nodes
- Added support for parsing for `FunctionCallOptions` (of the form `fn{key=value}(args)`)
- Added support for parsing `IndexRangeAccess`es (of the form `array[from:to]`)
- Added proper nodes for the `ReturnStatement`, `BreakStatement`, and `ContinueStatement` (which were previously rendered as the return expression or just `';'` for expression-less statements)
- Fixed parser discarding all variable declarations in a variable declaration list if one is `None`
- Fixed parser failing if an assembly member's identifier is a list (not quite sure where that was a problem TBH)
- Fixed parser failing an assembly assignment to a member

I added some more fixes with commit eb548d1 on 2022-05-12:

- Added `isDeclaredImmutable` to variable declaration, similar to `isDeclaredConst`
- Rendered omitted loop expressions in for loops as `None` instead of an `ExpressionStatement` with `None` as its expression
- Added `storageLocation` field to variable declarations inside a variable declaration list

florian1345 committed:
Various fixes/missing fields

- Added `name` field to `TypeDefinition` nodes
- Added `isVirtual` and `isOverride` fields to `ModifierDefinition` nodes
- Added support for parsing for `FunctionCallOptions` (of the form `fn{key=value}(args)`)
- Added support for parsing `IndexRangeAccess`es (of the form `array[from:to]`)
- Fixed parser discarding all variable declarations in a variable declaration list if one is `None`
- Fixed parser failing if an assembly member's identifier is a list
- Fixed parser failing an assembly assignment to a member
- Added proper nodes for the `ReturnStatement`, `BreakStatement`, and `ContinueStatement`

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/ConsenSys/python-solidity-parser?pullRequest=24) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/ConsenSys/python-solidity-parser?pullRequest=24) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/ConsenSys/python-solidity-parser?pullRequest=24) it.</sub>

florian1345 committed:
More fixes

- Added "isDeclaredImmutable" to variable declaration, similar to "isDeclaredConst"
- Rendered omitted loop expressions in for loops as None instead of an ExpressionStatement with None as its expression
- Added storageLocation field to variable declarations inside a variable declaration list

nodablock commented:
Would that be okay to review and merge this PR @tintinweb ?
